staterecorder.go

```diff
@@ -12,79 +12,132 @@ const (
 	CoreTerminated
 )
 
-// StateRecorder implements a Reporter which records the most recent operation
```

Comment:
TODO This file needs a review, has been AI generated and I touched it up only slightly.

Reply:
Tidied up.